Core/src/org/sleuthkit/autopsy/modules/hashdatabase/HashDbManager.java

```diff
@@ -494,14 +586,37 @@ private void configureSettings(HashLookupSettings settings) {
         will load without errors and the user may think that the problem was solved.*/
         if (!allDatabasesLoadedCorrectly && RuntimeProperties.runningWithGUI()) {
             try {
-                HashLookupSettings.writeSettings(new HashLookupSettings(this.knownHashSets, this.knownBadHashSets));
+                HashLookupSettings.writeSettings(new HashLookupSettings(HashLookupSettings.convertHashSetList(this.hashSets)));
                 allDatabasesLoadedCorrectly = true;
             } catch (HashLookupSettings.HashLookupSettingsException ex) {
                 allDatabasesLoadedCorrectly = false;
                 logger.log(Level.SEVERE, "Could not overwrite hash database settings.", ex);
             }
         }
     }
+    
+    void updateHashSetsFromCentralRepository() throws TskCoreException {
+        if(EamDb.isEnabled()){
+            List<HashDbInfo> crHashDbInfoList = getCentralRepoHashSetsFromDatabase();
+            for(HashDbInfo hashDbInfo : crHashDbInfoList) {
+                if(hashDbInfoIsNew(hashDbInfo)){
+                    addExistingCentralRepoHashSet(hashDbInfo.getHashSetName(), hashDbInfo.getVersion(), 
+                                hashDbInfo.getReferenceSetID(), 
+                                hashDbInfo.getSearchDuringIngest(), hashDbInfo.getSendIngestMessages(), hashDbInfo.getKnownFilesType(),
+                                hashDbInfo.isReadOnly());   
+                }
+            }
+        }
+    }
```

Comment:
I noticed that this method is being called from just one of the UI classes, and whenever I see one class controlling the behavior of another class, rather than asking for some service, I tend to wonder if implementation details have leaked into the API. Would it be better to have the HashDbManager refresh its view of the CR hash sets when it is queried for hash sets? It think that would be less error-prone for the clients of HashDbManager (they don't need to remember to call this method) and would reduce the scope of the overall class API (always shooting for minimal but sufficient). Thoughts?

Reply:
Assuming these close parentheses come from updateHashSetsFromCentralRepository, it is called from three places:
HashDbManager.getAllHashSets
HashDbManager.configureSettings
HashLookupModuleSettingsPanel.initializeHashSetModels

getAllHashSets covers most cases. We could add it in getKnownFileHashSets/getKnownBadFileHashSets and then the Panel wouldn't need to call it and it could be private.  It might make sense on getUpdateableHashSets, too.